### PR TITLE
Update L2geth from 0.5.2 to 0.5.8 

### PR DESCRIPTION
**Pull request by CAPtheorem**
Applies 12 Patches to L2Geth, bringing us back to sync with Optimism 0.5.8

### Patch Changes

- 949916f8: Add a better error message for when the sequencer url is not configured when proxying user requests to the sequencer for `eth_sendRawTransaction` when running as a verifier/replica
- 300f79bf: Fix nonce issue
- ae96d784: Add reinitialize-by-url command, add dump chain state command
- c7569a16: Fix blocknumber monotonicity logging bug

### Patch Changes

- d4bf299f: Add support to fully unmarshal Receipts with Optimism fields
- 8be69ca7: Add changeset for https://github.com/ethereum-optimism/optimism/pull/2011 - replicas forward write requests to the sequencer via a configured parameter `--sequencer.clienthttp` or `SEQUENCER_CLIENT_HTTP`
- c9fd6ec2: Correctly parse fee enforcement via config to allow turning off L2 fees for development

### Patch Changes

- 3a77bbcc: Implement updated timestamp logic
- 3e3c07a3: changed the default address to be address(0) in `call`

### Patch Changes

- 2924845d: expose ErrNonceTooHigh from miner

### Patch Changes

- d205c1d6: surface sequencer low-level sequencer execution errors

### Patch Changes

- 5febe10f: fixes empty block detection and removes empty worker tasks
- 272d20d6: renames l2geth package name to github.com/ethereum-optimism/optimism/l2geth

### CODE REVIEW

File: l2geth/eth/filters/api.go [hunk 1]
```diff
@@ -52,24 +52,22 @@ type filter struct {
...
 // information related to the Ethereum protocol such als blocks, transactions and logs.
 type PublicFilterAPI struct {
-	backend    Backend
-	mux        *event.TypeMux
-	quit       chan struct{}
-	chainDb    ethdb.Database
-	events     *EventSystem
-	filtersMu  sync.Mutex
-	filters    map[rpc.ID]*filter
-	rangeLimit bool
```
Comment (Boyuan-Chen) [inline l2geth/eth/filters/api.go:62]: @InoMurko added this for the `eth_getLogs`. Probably we should keep this?

File: l2geth/internal/ethapi/api.go [hunk 2]
```diff
@@ -1030,35 +1037,7 @@ func DoEstimateGas(ctx context.Context, b Backend, args CallArgs, blockNrOrHash
...
-
-	var data []byte
-	if args.Data != nil {
-		data = []byte(*args.Data)
-	}
-
-	msg := callmsg{types.NewMessage(*args.From, args.To, nonce, value, hi, gasPrice, data, false, new(big.Int), 0, []byte{0}, types.QueueOriginSequencer)}
-
-	state, _, _ := b.StateAndHeaderByNumberOrHash(ctx, blockNrOrHash)
-	l2ExtraGas := new(big.Int)
```
Comment (Boyuan-Chen) [inline l2geth/internal/ethapi/api.go]: This is for moving the l1 security fee to l2 gas.

**Review by Boyuan-Chen — CHANGES_REQUESTED** [review]

File: l2geth/eth/filters/api.go (hunk 1 above)
Comment (CAPtheorem) [reply l2geth/eth/filters/api.go:62]: Yup!

File: l2geth/README.md [hunk 3]
```diff
@@ -16,7 +16,7 @@ configuration will determine the mode of operation. The configuration flags
 can be configured using either environment variables or passed at runtime as
 flags.
 
-A prebuilt Docker image is available at `ethereumoptimism-optimism/optimism/l2geth`.
+A prebuilt Docker image is available at `ethereumoptimism/go-ethereum`.
```
Comment (InoMurko) [inline l2geth/README.md:19]: should we swap this for `bobanetwork/l2geth`?

Comment (CAPtheorem) [reply l2geth/README.md]: Please think about the best way to do this, longer term - not sure

**Comment by CAPtheorem** [comment]
completed

**Pull request closed without merging**